### PR TITLE
Enable component parsing.

### LINKED ISSUE #323: Define component type & re-enable component parsing in messages
See #322

### PR DESCRIPTION
**Pull request by jchristgit**
Define component type and re-enable parsing of message components.

Fixes #323.

### CODE REVIEW

**Comment by jchristgit** [comment]
Hi @jb3 or @Kraigie, does one of you have time to review this this weekend?

**Review by Th3-M4jor — CHANGES_REQUESTED** [review]
Some minor things in the typespec that need fixing, as well as a `?` in a name that shouldn't be present.

**Comment by jchristgit** [comment]
@Th3-M4jor the `?` was intentional to adhere to the Elixir "standard" of naming these fields. Should we keep it, or is this confusing?

**Comment by Th3-M4jor** [comment]
I'm of the opinion we should try to mirror the official api docs as much as possible with our field names.

**Comment by jchristgit** [comment]
Fair enough 👍

File: lib/nostrum/struct/message/component.ex [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+  Only present for action rows.
+  """
+  @type components :: [t()]
+
+  @typedoc "Represents a message component."
+  @type t :: %__MODULE__{
+          type: type(),
+          custom_id: custom_id(),
+          disabled?: disabled?(),
```
Comment (jchristgit) [inline lib/nostrum/struct/message/component.ex:166]:
```suggestion
          disabled: disabled(),
```

File: lib/nostrum/struct/message/component.ex [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+          components: components()
+        }
+
+  @doc false
+  @spec to_struct(map()) :: t()
+  def to_struct(map) do
+    %__MODULE__{
+      type: map.type,
+      custom_id: map[:custom_id],
+      disabled?: map[:disabled],
```
Comment (jchristgit) [inline lib/nostrum/struct/message/component.ex:184]:
```suggestion
      disabled: map[:disabled],
```

**Review by jb3 — APPROVED** [review]
LGTM! Some of the markdown links wrapping onto newlines is a bit peculiar but I get why you've done it.

**Comment by jb3** [comment]
Thanks!

**Pull request merged**